e2etests/tests/advertisement.go

```diff
@@ -0,0 +1,289 @@
+					}
+				},
+				validate: func(ppV4 []config.Peer, ppV6 []config.Peer, nodes []v1.Node) {
+					ValidatePrefixesForNeighbor(ppV6[0].FRR, nodes, "fc00:f853:ccd:e799::/64")
+					ValidateNeighborNoPrefixes(ppV6[0].FRR, nodes, "fc00:f853:ccd:e800::/64")
+					for _, p := range ppV6[1:] {
+						ValidatePrefixesForNeighbor(p.FRR, nodes, "fc00:f853:ccd:e799::/64", "fc00:f853:ccd:e800::/64")
+					}
+				},
+			}),
+			ginkgo.Entry("IPV4 - VRF - Advertise with mode allowall", params{
+				ipFamily: ipfamily.IPv4,
+				vrf:      infra.VRFName,
+				myAsn:    infra.FRRK8sASNVRF,
+				prefixes: []string{"192.168.2.0/24", "192.169.2.0/24"},
+				modifyPeers: func(ppV4 []config.Peer, ppV6 []config.Peer) {
+					for i := range ppV4 {
+						ppV4[i].Neigh.ToAdvertise.Allowed.Mode = frrk8sv1beta1.AllowAll
+					}
+				},
+				validate: func(ppV4 []config.Peer, ppV6 []config.Peer, nodes []v1.Node) {
+					for _, p := range ppV4 {
+						ValidatePrefixesForNeighbor(p.FRR, nodes, "192.168.2.0/24", "192.169.2.0/24")
+					}
+				},
+			}),
+			ginkgo.Entry("IPV4 - VRF - Advertise a subset of ips", params{
+				ipFamily: ipfamily.IPv4,
+				vrf:      infra.VRFName,
+				myAsn:    infra.FRRK8sASNVRF,
+				prefixes: []string{"192.168.2.0/24", "192.169.2.0/24"},
+				modifyPeers: func(ppV4 []config.Peer, ppV6 []config.Peer) {
+					ppV4[0].Neigh.ToAdvertise.Allowed.Prefixes = []string{"192.168.2.0/24"}
+					for i := 1; i < len(ppV4); i++ {
+						ppV4[i].Neigh.ToAdvertise.Allowed.Prefixes = []string{"192.168.2.0/24", "192.169.2.0/24"}
+					}
+				},
+				validate: func(ppV4 []config.Peer, ppV6 []config.Peer, nodes []v1.Node) {
+					ValidatePrefixesForNeighbor(ppV4[0].FRR, nodes, "192.168.2.0/24")
+					ValidateNeighborNoPrefixes(ppV4[0].FRR, nodes, "192.169.2.0/24")
+					for _, p := range ppV4[1:] {
+						ValidatePrefixesForNeighbor(p.FRR, nodes, "192.168.2.0/24", "192.169.2.0/24")
+					}
+				},
+			}),
+			ginkgo.Entry("IPV6 - Advertise a subset of ips", params{
+				ipFamily: ipfamily.IPv6,
+				vrf:      infra.VRFName,
+				myAsn:    infra.FRRK8sASNVRF,
+				prefixes: []string{"fc00:f853:ccd:e799::/64", "fc00:f853:ccd:e800::/64"},
+				modifyPeers: func(ppV4 []config.Peer, ppV6 []config.Peer) {
+					ppV6[0].Neigh.ToAdvertise.Allowed.Prefixes = []string{"fc00:f853:ccd:e799::/64"}
+					for i := 1; i < len(ppV6); i++ {
+						ppV6[i].Neigh.ToAdvertise.Allowed.Prefixes = []string{"fc00:f853:ccd:e799::/64", "fc00:f853:ccd:e800::/64"}
+					}
+				},
+				validate: func(ppV4 []config.Peer, ppV6 []config.Peer, nodes []v1.Node) {
+					ValidatePrefixesForNeighbor(ppV6[0].FRR, nodes, "fc00:f853:ccd:e799::/64")
+					ValidateNeighborNoPrefixes(ppV6[0].FRR, nodes, "fc00:f853:ccd:e800::/64")
+					for _, p := range ppV6[1:] {
+						ValidatePrefixesForNeighbor(p.FRR, nodes, "fc00:f853:ccd:e799::/64", "fc00:f853:ccd:e800::/64")
+					}
+				},
+			}),
+			ginkgo.Entry("DUAL - Advertise with mode allowall", params{
```

Comment:
`DUAL` -> `DUALSTACK` as that is what we skip in the ci? (although this passes on these lanes)